src/Roundabout/roundabout-it-button.tsx

```diff
@@ -111,26 +116,31 @@ export function RoundaboutItButton({
         },
         [iteration, goToIteration],
     );
-
+    const { css } = useStyles();
+    console.log(unnecessary ? "true" : false);
+    if (unnecessary) {
+        return null;
+    }
     return (
-        <div className="fr-col">
-            <div className="fr-grid-row">
+        <div className="fr-col-12 fr-col-md-6">
+            <div
+                className={css({
+                    [fr.breakpoints.up("md")]: {
+                        justifyContent: "flex-end",
+                        display: "flex",
+                    },
+                })}
+            >
                 <CompleteBadge status={status} locked={locked} />
-                <div className="fr-col-12">
-                    <DisplayButton
-                        status={status}
-                        locked={locked}
-                        onClick={onClick}
-                        custom={custom}
-                        label={label}
-                        unnecessary={unnecessary}
-                    />
-                </div>
+                <DisplayButton
+                    status={status}
+                    locked={locked}
+                    onClick={onClick}
+                    custom={custom}
+                    label={label}
+                    unnecessary={unnecessary}
```

Comment:
I don't think you need to pass uncessqry here

Reply:
I think we do: it's preferable to avoid including if/else logic within the return statement. And according to the figma, the button should disappear when the `unnecessary` expression returns a string.

Reply:
Because we have two options, either we insert the `if (unnecessary)` in the return statement of `RoundaboutIteration`: 

 ```js
function RoundaboutIteration({unnecessary}) {
    return (
        <RoundaboutItContainer>
            <RoundaboutItTitle/>
            {unnecessary && 
            <RoundaboutItButton/>}
        </RoundaboutItContainer>
    );
}
 ```
 Or we pass `unnecessary` to `RoundaboutItButton` and do an early return of null if unnecessary is true-y (i.e. if a string is present). 
 
In Stromae/Lunatic-DSFR, we decided to use systematically the second strategy because we want to extract all logic from the return statement and construct all components with building blocks.

Reply:
To close the loop on this, there were two  `if (unnecessary) return null;`, one nested within the other. 

The nested one, as you pointed out, can be removed, as it would not be called in any case.